hugr-py/src/hugr/tys.py

```diff
@@ -309,6 +338,9 @@ class FunctionType(Type):
     output: TypeRow
     extension_reqs: ExtensionSet = field(default_factory=ExtensionSet)
 
+    def type_bound(self) -> TypeBound:
+        return TypeBound.Copyable
```

Comment:
Is this always the case? Could we have some partially-applied function with linear state?

Reply:
I just copied the spec/crate.
The justification I believe is that if the linear state can be encoded in to the function then it must be copyable, so there's no need to mark it as a linear type. It is a bit restrictive, you may wish to tell the type system to treat something as linear even if you can encode it. This is one for a discussion or issue I think.